library/Asterisk/ARI/Connector.php

```diff
@@ -0,0 +1,109 @@
+
+        // Get Asterisk credetentials from application configuration
+        $bootstrap = \Zend_Controller_Front::getInstance()->getParam('bootstrap');
+        if (is_null($bootstrap)) {
+            $conf = new \Zend_Config_Ini(APPLICATION_PATH . '/configs/application.ini', APPLICATION_ENV);
+            $config = (Object) $conf->toArray();
+        } else {
+            $config = (Object) $bootstrap->getOptions();
+        }
+
+        $this->_user = $config->ari["userName"];
+        $this->_pass = $config->ari["password"];
+        $this->_port = $config->ari["port"];
+    }
+
+    private function _createConnectionHandler()
+    {
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_USERPWD, $this->_user . ":" . $this->_pass);
+        curl_setopt($ch, CURLOPT_POST, TRUE);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array("Content-Type: application/json"));
+        return $ch;
+    }
+
+    private function _post($ch, $url, $postdata)
+    {
+        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($postdata));
+        curl_setopt($ch, CURLOPT_URL, $url);
+        return curl_exec($ch);
+    }
+
+    public function sendFaxfileRequest(\IvozProvider\Model\Raw\FaxesInOut $faxfile)
+    {
+        $applicationSerersMapper = new \IvozProvider\Mapper\Sql\ApplicationServers();
+        $applicationServer = $applicationSerersMapper->fetchList();
+
+        // Try sending the request to all existing Application Servers secuentially
+        foreach ($applicationServer as $as) {
+            $destination = $faxfile->getDst();
+            $asAddress = $as->getIp();
+            $company = $faxfile->getFax()->getCompany();
+
+            // Get a new connection handler for this application server
+            $ch = $this->_createConnectionHandler();
+
+            // Setup POST body content
+            $post = array(
+                    "endpoint"  => "Local/$destination@faxes",
+                    "context"   => "faxes-send",
+                    "extension" => $destination,
+                    "variables" => array(
+                            "__LOGTAG"    => "b" . $company->getBrandId(),
+                            "__CID_HASH"  => "faxout" . $faxfile->getId(),
+                            "__FAXFILE_ID" => (string) $faxfile->getId(),
+                    )
+            );
+
+            // Get Application server restAPI URL
+            $url = "http://" . $asAddress . ":" . $this->_port . "/ari/channels";
+            $this->_logger->log("Sending ARI Request to $url...", \Zend_Log::INFO);
+
+            // Reques a new call
+            $response = $this->_post($ch, $url, $post);
+            if ($response === false) {
+                $this->_logger->log("Error sending fax to $url:" . curl_error($ch), \Zend_Log::ERR);
+            } else {
+                $this->_logger->log("Request successfully enqueued.", \Zend_Log::INFO);
```

Comment:
If I disable this option to make curl_exec return boolean:
```
// curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
```
Maybe something like this?

```
    $response = $this->_post($ch, $url, $post);
    if ($response === true) {
        $this->_logger->log("Request successfully enqueued to $url.", \Zend_Log::INFO);
        break;
    }
    $this->_logger->log("Error sending fax to $url:" . curl_error($ch), \Zend_Log::ERR);
} /* foreach loop */

return $response;

```